pkg/controller/operators/catalog/operator.go

```diff
@@ -1062,6 +1142,15 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 		logger.Debugf("no subscriptions were updated")
 	}
 
+	// Make sure that we no longer indicate unpacking progress
+	subs = o.setSubsCond(subs, v1alpha1.SubscriptionCondition{
+		Type:   v1alpha1.SubscriptionBundleUnpacking,
+		Status: corev1.ConditionFalse,
+	})
+
+	// Remove BundleUnpackFailed condition from subscriptions
+	subs = o.removeSubsCond(subs, v1alpha1.SubscriptionBundleUnpackFailed)
```

Comment:
Ok, I found a bug with this line. `o.removeSubsCond` only returns a list of subs which were updated.

After hitting this line in successfull installation from the first time - we will have `nil` in `subs` aftere this line. And it might result in `ConstraintsNotSatisfiable` or other conditions to not be updated.

Reply:
We now do `o.removeSubsCond(subs, v1alpha1.SubscriptionBundleUnpackFailed)` and `o.removeSubsCond(subs, v1alpha1.SubscriptionResolutionFailed)` (not overriding `subs` slice).

I believe `v1alpha1.SubscriptionResolutionFailed` had the same issue.